style(components): [select] Modify element hierarchy

ma-shuo opened the pull request:
Modify element hierarchy to support input right-click paste function.

closed #17886

Please make sure these boxes are checked before submitting your PR, thank you!

- [x] Make sure you follow contributing guide [English](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.en-US.md) | ([中文](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.zh-CN.md) | [Español](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.es.md) | [Français](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.fr-FR.md)).
- [x] Make sure you are merging your commits to `dev` branch.
- [x] Add some descriptions and refer to relative issues for your PR.

ma-shuo committed:
style(components): [select] Modify element hierarchy

Modify element hierarchy to support input right-click paste function.

closed #17886

ma-shuo committed:
style(components): [select] Modify element hierarchy

Modify element hierarchy to support input right-click paste function.

closed #17886

tolking commented:
<img width="277" alt="截屏2024-08-14 16 02 55" src="https://github.com/user-attachments/assets/fec77603-4f56-4704-b342-bb3368dd08cb">

ma-shuo committed:
Merge remote-tracking branch 'origin/dev-style1' into dev-style1

# Conflicts:
#	packages/theme-chalk/src/select.scss

tolking commented on packages/components/select/src/select.vue:
```diff
@@ -156,7 +156,6 @@
                 :class="[nsSelect.e('input'), nsSelect.is(selectSize)]"
                 :disabled="selectDisabled"
                 :autocomplete="autocomplete"
-                :style="inputStyle"
```
It seems that `inputStyle` is only used here, and perhaps the related code can be removed at the same time.

In addition, since the Select component and the SelectV2 component use the same style, it may be necessary to handle the SelectV2 component at the same time.

ma-shuo committed:
fix(components): [select] Modify element hierarchy

Modify element hierarchy to support input right-click paste function.

closed #17886

ma-shuo committed:
Merge branch 'dev' into dev-style1

ma-shuo committed:
Merge branch 'dev' into dev-style1

btea commented:
Please refer to the error warning message, the variable `MINIMUM_INPUT_WIDTH'` will also be removed.

ma-shuo committed:
fix(components): [select] Modify element hierarchy

closed #17886

ma-shuo committed:
Merge remote-tracking branch 'origin/dev-style1' into dev-style1

ma-shuo commented:
> 请参考错误警告信息，该变量`MINIMUM_INPUT_WIDTH'`也将被删除。

ok

ma-shuo committed:
Merge branch 'dev' into dev-style1

kooriookami commented:
![image](https://github.com/user-attachments/assets/7b39607f-50ae-4098-832c-9a4c69f286f9)

tolking commented:
> I checked the element documentation for Vue2 and found that the style is also the same, so I'm not sure if this is correct

We need to maintain consistency with the previous version. Perhaps we can fix this issue by changing the `width` property of `inputStyle` to the `minWidth`.

ma-shuo committed:
Merge branch 'dev' into dev-style1

ma-shuo commented:
<img width="480" alt="image" src="https://github.com/user-attachments/assets/4abacdbd-a208-4a7f-9cab-3119eba8d5f5">
Sorry, it's been a bit long. I tested using 'min width' and found that this situation occurs

ma-shuo commented:
> > 我检查了 Vue2 的元素文档，发现样式也是相同的，所以我不确定这是否正确
> 
> 我们需要与之前的版本保持一致。也许我们可以通过将 的属性更改为 来解决这个`width`问题。`inputStyle``minWidth`

<img width="926" alt="image" src="https://github.com/user-attachments/assets/62e23e5d-c115-4401-bc4f-cdbad0601886">

Sorry, it's been a bit long. I tested using 'min width' and found that this situation occurs.

ma-shuo commented:
> > I checked the element documentation for Vue2 and found that the style is also the same, so I'm not sure if this is correct
> 
> We need to maintain consistency with the previous version. Perhaps we can fix this issue by changing the `width` property of `inputStyle` to the `minWidth`.

<img width="926" alt="image" src="https://github.com/user-attachments/assets/62e23e5d-c115-4401-bc4f-cdbad0601886">

Sorry, it's been a bit long. I tested using 'min width' and found that this situation occurs.

ma-shuo committed:
style(components): modify element hierarchy

closed #17886

tolking committed:
fix(components): [select] change the width of the input to the minWidth

tolking commented:
Looks pretty normal, cc @ma-shuo

ma-shuo committed:
Merge branch 'dev' into dev-style1

ma-shuo commented:
> 看起来很正常，cc@ma-shuo

如果使用 'min-width' 的话无论在什么情况下输入内容的时候都会换行，这显然是不合理的

tolking commented:
You can try it first [demo](https://preview-17889-element-plus.surge.sh/en-US/component/select.html#create-new-items)

![2024-10-28 10 47 46](https://github.com/user-attachments/assets/5d11b6ac-af8b-422c-865b-68e6945aecda)

ma-shuo committed:
Merge branch 'element-plus:dev' into dev-style1
